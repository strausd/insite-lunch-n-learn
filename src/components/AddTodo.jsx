import React from 'react';
import { connect } from 'react-redux';

import { addTodo } from '../actions/todosActions';


class AddTodo extends React.Component {

    constructor(props) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(e) {
        e.preventDefault();
        const todoText = this.refs.todoText.value;
        this.props.addTodo(todoText);
        this.refs.todoText.value = '';
    }

    render() {
        return (
            <div>
                <h2>Add a todo</h2>
                <form onSubmit={this.handleSubmit}>
                    <input name="add-todo" id="add-todo" placeholder="Todo description" ref="todoText" />
                </form>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addTodo: (name) => dispatch(addTodo(name))
    };
};

export default connect(undefined, mapDispatchToProps)(AddTodo);