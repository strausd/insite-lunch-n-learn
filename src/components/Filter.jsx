import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';

import { setText, filterCompleted, filterDate } from '../actions/filterActions';


class Filter extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            focusedInput: null,
            startDate: null,
            endDate: null
        };
        this.props.setFilterText('');
        this.handleFilterText = this.handleFilterText.bind(this);
        this.handleFilterComplete = this.handleFilterComplete.bind(this);
        this.handleDateChange = this.handleDateChange.bind(this);
    }

    handleFilterText() {
        this.props.setFilterText(this.refs.filterText.value);
    }

    handleFilterComplete() {
        this.props.filterCompleted();
    }

    handleDateChange({ startDate, endDate }) {
        this.setState({ startDate, endDate });
        this.props.filterDate(startDate, endDate);
    }

    render() {
        return (
            <div>
                <input
                    name="filterText"
                    ref="filterText"
                    placeholder="Filter"
                    onChange={this.handleFilterText}
                />
                <label>Toggle Show Completed</label>
                <input
                    name="filterComplete"
                    ref="filterComplete"
                    type="checkbox"
                    onChange={this.handleFilterComplete}
                    checked={this.props.showCompleted}
                />
                <DateRangePicker
                    startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                    startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                    endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                    endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                    onDatesChange={ this.handleDateChange } // PropTypes.fungggc.isRequired,
                    focusedInput={ this.state.focusedInput } // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                    onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setFilterText: (text) => dispatch(setText(text)),
        filterCompleted: () => dispatch(filterCompleted()),
        filterDate: (start, end) => dispatch(filterDate(start, end))
    };
};

const mapStateToProps = (state) => {
    return {
        showCompleted: state.filters.showCompleted
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Filter);