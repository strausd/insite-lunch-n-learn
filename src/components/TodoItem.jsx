import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';

import { deleteTodo, toggleCompletedTodo } from '../actions/todosActions';


export class TodoItem extends React.Component {
    constructor(props) {
        super(props);
        this.handleDeleteTodo = this.handleDeleteTodo.bind(this);
        this.handleToggleComplete = this.handleToggleComplete.bind(this);
    }

    handleDeleteTodo() {
        this.props.deleteTodo(this.props.todo.id);
    }

    handleToggleComplete() {
        this.props.toggleCompleted(this.props.todo.id);
    }

    render() {
        return (
            <div>
                <p className={this.props.todo.completed ? 'complete' : 'not-complete'}>
                    {this.props.todo.name} - {this.props.todo.completed ? 'Completed' : 'Not Complete'} - {moment(this.props.todo.createdAt).format("MM/DD/YY")}
                </p>
                <button onClick={this.handleDeleteTodo}>Delete</button>
                <button onClick={this.handleToggleComplete}>Toggle Complete</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        deleteTodo: (id) => {
            dispatch(deleteTodo(id));
        },
        toggleCompleted: (id) => {
            dispatch(toggleCompletedTodo(id));
        }
    };
};

export default connect(undefined, mapDispatchToProps)(TodoItem);