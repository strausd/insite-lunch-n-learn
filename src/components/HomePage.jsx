import React from 'react';
import { connect } from 'react-redux';

import TodoList from './TodosList';
import AddTodo from './AddTodo';


export class HomePage extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                <AddTodo />
                <TodoList />
            </div>
        );
    }
};

export default HomePage;