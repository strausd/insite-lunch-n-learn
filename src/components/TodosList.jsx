import React from 'react';
import { connect } from 'react-redux';

import TodoItem from './TodoItem'


class TodoList extends React.Component {


    render() {
        return (
            <div>
                <h2>Todo List component</h2>
                {this.props.todos.map((todo) => <TodoItem key={todo.id} todo={todo} />)}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        todos: state.todos
    };
}

export default connect(mapStateToProps)(TodoList);