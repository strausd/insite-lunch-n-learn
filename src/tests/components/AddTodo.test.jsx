import React from 'react';
import { shallow, mount } from 'enzyme';

import { AddTodo } from '../../components/AddTodo';
import { todos } from '../mockdata/todos';


let wrapper, addTodoSpy;

beforeAll(() => {
    addTodoSpy = jest.fn();
    wrapper = shallow(<AddTodo addTodo={ addTodoSpy } />);
});

test('should render AddTodo component properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should submit form with input value for new todo item', () => {
    const mounted_wrapper = mount(<AddTodo addTodo={addTodoSpy} />);
    const new_value = 'ksjdhgkjsh';
    mounted_wrapper.find('input').at(0).instance().value = new_value;
    mounted_wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    expect(addTodoSpy).toHaveBeenLastCalledWith(new_value);
});
