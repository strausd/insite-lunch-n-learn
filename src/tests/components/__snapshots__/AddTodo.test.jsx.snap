// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddTodo component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTodo
    addTodo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Add a todo
        </h2>,
        <form
          onSubmit={[Function]}
        >
          <input
            id="add-todo"
            name="add-todo"
            placeholder="Todo description"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a todo",
        },
        "ref": null,
        "rendered": "Add a todo",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            id="add-todo"
            name="add-todo"
            placeholder="Todo description"
          />,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "add-todo",
            "name": "add-todo",
            "placeholder": "Todo description",
          },
          "ref": "todoText",
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Add a todo
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <input
              id="add-todo"
              name="add-todo"
              placeholder="Todo description"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a todo",
          },
          "ref": null,
          "rendered": "Add a todo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              id="add-todo"
              name="add-todo"
              placeholder="Todo description"
            />,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "add-todo",
              "name": "add-todo",
              "placeholder": "Todo description",
            },
            "ref": "todoText",
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
